yaml_edit/lib/src/editor.dart

```diff
@@ -449,7 +436,7 @@ class YamlEditor {
           updatedYamlMap(parentNode, (nodes) => nodes.remove(keyOrIndex));
     }
 
-    _performEdit(edit, collectionPath, expectedNode);
+    _performEdit(edit ?? SourceEdit(0, 0, ''), collectionPath, expectedNode);
```

Comment:
`edit` cannot be `null` here can it?

we can simply add an `else` handler here..
Because if `parentNode` is not a `YamlList` or `YamlMap`, then we have a case like:
```yaml
key:
  subkey: my-string
```
If I was to do `editor.remove(['key', 'subkey', 'subsubkey']);` then we should throw a `PathError` because `subkey` is a string, and we cannot remove the `subsubkey` from a string...

-----

We probably need a test case for this too.

Reply:
Sorry this was me being unsure how `edit` can be guaranteed to have a value. I could have done:

```dart
    _performEdit(edit!, collectionPath, expectedNode);
```
...and no tests would fail. But that didn't seem right. You're probably correct though that this will always be instantiated.

What should be the value of `edit` for the else case?

About the test scenario, that's covered under `test/remove_test.dart` on line 35 "PathError if collectionPath points to a scalar".

Reply:
I would suggest making `edit` a `late SourceEdit edit`, but that's a nit :D